### PR TITLE
COR-332: add month field type

### PR DESCRIPTION
**Pull request by shinroo**
- Adds a new field type "Month", which is a combination of a year and a month in the `YYYY-MM` format (this uniquely identifies a month)
- Validation with a `pattern` attribute on the input

### CODE REVIEW

File: web/pwa/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "axios": "^0.23.0",
     "bootstrap": "^5.1.3",
     "bootstrap-icons": "^1.6.1",
+    "inputmask": "^5.0.6",
```
Comment (ludydoo) [inline web/pwa/package.json:20]: Is this still needed ?

**Review by ludydoo — CHANGES_REQUESTED** [review]

File: web/pwa/src/features/recorder/FieldEditor.tsx [hunk 2]
```diff
@@ -71,6 +71,43 @@ export const DateFieldEditor: FC<FieldEditorProps> = props => {
...
+
+    const [localValue, setLocalValue] = useState(value != null ? (value as Date).toISOString().slice(0, 7) : "")
+
+    const isValidLength = () => localValue.length === expectedLength;
+
+    function isValid(s: string) {
+        const valid = /^(?:19|20|21)\d{2}-[01]\d$/
+        const m = +s.slice(5)
+        return valid.test(s) && 0 < m && m <= 12;
+    }
```
Comment (ludydoo) [inline web/pwa/src/features/recorder/FieldEditor.tsx:86]: Same comment here about dateFns..

**Review by ludydoo — CHANGES_REQUESTED** [review]

File: web/pwa/package.json (hunk 1 above)
Comment (shinroo) [reply web/pwa/package.json]: Removed it good catch!

File: pkg/sqlconvert/convert.go [hunk 3]
```diff
@@ -174,6 +175,12 @@ func convertFieldToSqlField(formDef *types.FormDefinition, field *types.FieldDef
 		result.DataType = sqlschema.SQLDataType{
 			Date: &sqlschema.SQLDataTypeDate{},
 		}
+	} else if field.FieldType.Month != nil {
+		result.DataType = sqlschema.SQLDataType{
+			VarChar: &sqlschema.SQLDataTypeVarChar{
```
[the remark this answers is not in the document]
Comment (shinroo) [reply pkg/sqlconvert/convert.go]: @nilueps and I implemented this yesterday

File: web/pwa/src/features/recorder/FieldEditor.tsx (hunk 2 above)
Comment (shinroo) [reply web/pwa/src/features/recorder/FieldEditor.tsx]: I've kept the isValid function in place, but replaced the serialisiation with the format function from `date-fns`

**Review by ludydoo — APPROVED** [review]

**Pull request merged**